components/salesforce/common.js

```diff
@@ -1,152 +1,77 @@
   },
   hooks: {
     async activate() {
-      // Retrieve base API URL's for this account
-      const apiUrls = await this.salesforce.getApiUrls();
-      this.db.set("apiUrls", apiUrls);
-
-      // Enable this Salesforce account to make HTTP
-      // calls to this event source's endpoint.
-      // API docs: https://sforce.co/3jAeV0G
-      const { metadata: metadataApiUrl } = apiUrls;
-      const remoteSiteName = this._getRemoteSiteName();
-      const { endpoint: endpointUrl } = this.http;
-      await this.salesforce.createRemoteSite(metadataApiUrl, remoteSiteName, endpointUrl);
-      this.db.set("remoteSiteName", remoteSiteName);
+      const latestDateCovered = this.db.get("latestDateCovered");
+      if (!latestDateCovered) {
+        const now = new Date().toISOString();
+        this.db.set("latestDateCovered", now);
+      }
 
-      // Create a custom Apex class to handle the HTTP
-      // request logic of the webhook.
-      const secretToken = uuidv4();
-      const webhookName = this._getWebhookName();
-      const webhookBody = this.getWebhookBody(webhookName, secretToken);
-      const { id: webhookId } = await this.salesforce.createApexClass(webhookName, webhookBody);
-      this.db.set("secretToken", secretToken);
-      this.db.set("webhookId", webhookId);
-
-      // Create a custom Apex trigger to listen to
-      // the relevant events and dispatch the webhook
-      // call(s) accordingly.
-      const triggerName = this._getTriggerName();
-      const triggerBody = this.getTriggerBody(triggerName, webhookName);
-      const objectType = this.getObjectType();
-      const { id: triggerId } =
-        await this.salesforce.createApexTrigger(triggerName, triggerBody, objectType);
-      this.db.set("triggerId", triggerId);
-    },
-    async deactivate() {
-      const triggerId = this.db.get("triggerId");
-      await this.salesforce.deleteApexTrigger(triggerId);
-
-      const webhookId = this.db.get("webhookId");
-      await this.salesforce.deleteApexClass(webhookId);
-
-      const { metadata: metadataApiUrl } = this.db.get("apiUrls");
-      const remoteSiteName = this.db.get("remoteSiteName");
-      await this.salesforce.deleteRemoteSite(metadataApiUrl, remoteSiteName);
+      const nameField = await this.salesforce.getNameFieldForObjectType(this.objectType);
+      this.db.set("nameField", nameField);
     },
   },
   methods: {
-    _getRemoteSiteName() {
-      const eventSourceName = this.getEventSourceName();
-      return this.standardizeName(eventSourceName);
-    },
-    _getTriggerName() {
-      return this.standardizeName("trigger");
-    },
-    _getWebhookName() {
-      return this.standardizeName("webhook");
-    },
-    _isValidSource(event) {
-      const {
-        "x-webhook-token": webhookToken,
-      } = event.headers;
-      const secretToken = this.db.get("secretToken");
-      return webhookToken === secretToken;
+    isValidSObject(sobject) {
+      return sobject.replicateable;
```

Comment:
Could you describe what the function of the `replicateable` field is briefly in a comment in this method?

Reply:
Will do